### DISCUSSION TITLE
Kibana/Elasticstack is http not https

### DISCUSSION THREAD

**Discussion by mtnsec** [Security-Onion-Solutions/securityonion#3615] [Unsupported Versions] [answered]
I previously asked about this here:  https://github.com/Security-Onion-Solutions/securityonion/discussions/3454

I converted to the Features version of Elasticstack back when I was on SO 2.3.21, and accepted the warning that it was going to disable encryption. However, I do want my Elasticstack to be encrypted, and was looking forward to that being default in 2.3.40.

It appears that the upgrade did not implement TLS seamlessly after all, so my original question in my last post stands.  How do I get to TLS?  Is the setup script fixed yet so that I can run it again without having to do a full reinstall of the operating system and all?  (I'm on Ubuntu 18.04)

**UPDATE** - Originally I was getting 404 errors on https:/manager/kibana.  Now after waiting, it's been saying "Kibana server is not ready yet" so I have hope that https is actually implemented.

How do I check the status of kibana?  so-status says everything has started..

**UPDATE 2** - 2 hours later and Kibana still says "Kibana server is not ready yet".  Any ideas how to troubleshoot this?

**Comment by TOoSmOotH** [comment]
Did you look at the Kibana log? Most likely the issue is the kibana index migration.

**Reply by mtnsec** [reply]
Edit - Nevermind, found it.  And I see the error and am trying the fix now: https://docs.securityonion.net/en/2.3/kibana.html?highlight=kibana#diagnostic-logging

**Comment by mtnsec** [comment]
Okay after deleting the indexes as suggested in the documentation, I have new errors.

so-restart-kibana has this error/message repeated:
```

                  Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (300/300)
                  Server is not ready
                  Server still not ready after 300 attempts; giving up.
```

kibana.log repeats this over and over:

`{"type":"log","@timestamp":"2021-03-24T19:00:09+00:00","tags":["warning","plugins","licensing"],"pid":293,"message":"License information could not be obtained from Elasticsearch due to Error: Cluster client cannot be used after it has been closed. error"}
`

And it still says Kibana is not ready but the index error is gone.

**Reply by TOoSmOotH** [reply]
Try restarting elastic then restarting kibana

**Reply by mtnsec** [reply]
I ran sudo so-elastic-restart and it restarted every elastic component in sequence.

First, elasticsearch was restarted, no errors.

Then kibana was restarted, same error: 

```
                  Waiting for value 'Elastic' at 'http://localhost:5601/app/kibana' (300/300)
                  Server is not ready
                  Server still not ready after 300 attempts; giving up.
```

Then it restarted logstash, filebeat, curator, elastalert, etc.  And no errors for any of those either. Seems to be only Kibana has errors.

For good measure, I ran so-kibana-restart again after, same thing.  Attached are the full output of both commands --

[so-elastic-restart.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6200960/so-elastic-restart.txt)
[so-kibana-restart.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6200961/so-kibana-restart.txt)

**Reply by TOoSmOotH** [reply]
Have you put anything in the local directory before?

**Reply by mtnsec** [reply]
No, I haven't put anything new into the local directory - but I did make a change to the global.sls to fix the docker 172.17.0.0/24/24 network problem I had earlier today.

I see this again in my kibana.log:

```
{"type":"log","@timestamp":"2021-03-24T23:35:16+00:00","tags":["warning","savedobjects-service"],"pid":293,"message":"Another Kibana instance appears to be migrating the index. Waiting for that migration to complete. If no other Kibana instance is attempting migrations, you can get past this message by deleting index .kibana_3 and restarting Kibana."}
{"type":"log","@timestamp":"2021-03-24T23:35:17+00:00","tags":["info","savedobjects-service"],"pid":293,"message":"Migrating .kibana_task_manager_1 saved objects to .kibana_task_manager_2"}
```

These are new logs about the same two indexes I deleted previously, but they appear to have come back.

I just deleted them again:

```
mtn@so-lab:~$ curl -k -XDELETE https://localhost:9200/.kibana_task_manager_2
{"acknowledged":true}
mtn@so-lab:~$ curl -k -XDELETE https://localhost:9200/.kibana_3
{"acknowledged":true}
```
And restarted elastic again, but no change. :(

**Reply by TOoSmOotH** [reply]
ls -laR /opt/so/saltstack/local/salt/elasticsearch/

ls -laR /opt/so/saltstack/local/salt/kibana/

**Reply by mtnsec** [reply]
```
root@so-lab:~# ls -laR /opt/so/saltstack/local/salt/elasticsearch/
/opt/so/saltstack/local/salt/elasticsearch/:
total 16
drwxr-xr-x  4 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 49 socore socore 4096 Dec 13 17:55 ..
drwxr-xr-x  4 socore socore 4096 Dec 13 17:55 files
drwxr-xr-x  4 socore socore 4096 Dec 13 17:55 templates

/opt/so/saltstack/local/salt/elasticsearch/files:
total 16
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 ..
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 ingest
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 scripts

/opt/so/saltstack/local/salt/elasticsearch/files/ingest:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 ..

/opt/so/saltstack/local/salt/elasticsearch/files/scripts:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 ..

/opt/so/saltstack/local/salt/elasticsearch/templates:
total 16
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 ..
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 custom
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 so

/opt/so/saltstack/local/salt/elasticsearch/templates/custom:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 ..

/opt/so/saltstack/local/salt/elasticsearch/templates/so:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 4 socore socore 4096 Dec 13 17:55 ..

root@so-lab:~# ls -laR /opt/so/saltstack/local/salt/kibana/
/opt/so/saltstack/local/salt/kibana/:
total 24
drwxr-xr-x  6 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 49 socore socore 4096 Dec 13 17:55 ..
drwxr-xr-x  2 socore socore 4096 Dec 13 17:55 bin
drwxr-xr-x  2 socore socore 4096 Dec 13 17:55 custom
drwxr-xr-x  2 socore socore 4096 Dec 13 17:55 etc
drwxr-xr-x  2 socore socore 4096 Dec 13 17:55 files

/opt/so/saltstack/local/salt/kibana/bin:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 6 socore socore 4096 Dec 13 17:55 ..

/opt/so/saltstack/local/salt/kibana/custom:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 6 socore socore 4096 Dec 13 17:55 ..

/opt/so/saltstack/local/salt/kibana/etc:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 6 socore socore 4096 Dec 13 17:55 ..

/opt/so/saltstack/local/salt/kibana/files:
total 8
drwxr-xr-x 2 socore socore 4096 Dec 13 17:55 .
drwxr-xr-x 6 socore socore 4096 Dec 13 17:55 ..

```

**Comment by mtnsec** [comment, marked as answer]
So, what would be the worst case scenario if I deleted more than just the indexes that keep reappearing in the logs and tried just deleting all of them and starting over?

`curl -k -XDELETE https://localhost:9200/.kibana_task*`

or even

`curl -k -XDELETE https://localhost:9200/.kibana*`

**Reply by mtnsec** [reply]
I tried 

`curl -k -XDELETE https://localhost:9200/.kibana_task*`

And it seems to be working now.  I haven't noticed any adverse effects from just deleting all of the tasks..  Fingers crossed, but it seems like I'm back up and running!
